discovery_engine_core/web-api2/src/server.rs

```diff
@@ -68,12 +68,14 @@ where
         let addr = config.net.bind_to;
         init_tracing(config.as_ref());
 
+        let json_config = JsonConfig::default().limit(config.net.max_body_size);
         let app_state = AppState::create(config).await?;
         let app_state = web::Data::new(app_state);
 
         HttpServer::new(move || {
             App::new()
                 .app_data(app_state.clone())
+                .app_data(json_config.clone())
```

Comment:
Not that clone is wrong, but isn't to_owned more idiomatic Rust?

Reply:
no, `to_owned()` is ideomatic for converting types like `str` to their "owned" representation (`String`). I.e. it's meant for cases where `clone` doesn't work because you are not simply cloning the type but converting it to a different type.